### PR TITLE
Prebid 7: Freestar Bid Adapter initial release & Deprecate Sortable Bid Adapter

### PR DESCRIPTION
**Pull request by nicgallardo**
## Type of change
- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
- Removal of Sortable Bid Adapter
- Addition of Freestar Bid Adapter
- test parameters for validating bids
```
{
  "bidder": "freestar",
    "params": {
      "siteId":"example.com",
      "tagId":"example_728x90_970x90_320x50",
      "keywords":{}
    }
}
```

- maintainer - prebid@freestar.com
- [x] official adapter submission

Accompanying [Documentation](https://github.com/prebid/prebid.github.io/pull/3688)

### CODE REVIEW

**Comment by bretg** [comment]
@nicgallardo - you can remove sortable from the docs, but it's considered a breaking change to remove it from the code repo. Please restore the .js files and explain in the .md files that the sortable bidder and analytics adapter are deprecated. You can remove them formally in PBJS 7.0, which is next month!

**Comment by nicgallardo** [comment]
@bretg I am attempting to merge our branch to the `prebid-7` branch so that it might be in the 7.0 release. If this is not the correct manner to add to that release please point me in the correct direction to add to the 7.0 release. Thanks!

>

**Comment by bretg** [comment]
Thanks Nic. Adding @dgirardi for guidance on how to get this change into 7, and @patmmccann to consider for  the release notes (remove sortable bid and analytics adapters)

**Comment by dgirardi** [comment]
This PR looks correct for getting this into 7; you will need to post a separate PR if you also want it in 6 - I suggest waiting until this gets approved.

**Comment by patmmccann** [comment]
tagging #7796 for linkage, thanks!

File: modules/freestarBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    const floor = getBidFloor(bid);
+    const validFloor = !floor || isNumber(floor);
+    const validKeywords = !bid.params.keywords ||
+      (isPlainObject(bid.params.keywords) &&
+        Object.keys(bid.params.keywords).every(key =>
+          isStr(key) && isStr(bid.params.keywords[key])
+        ))
+    const isBanner = !bid.mediaTypes || bid.mediaTypes[BANNER];
+    const bannerSizes = isBanner ? deepAccess(bid, `mediaType.${BANNER}.sizes`) || bid.sizes : null;
+    return !!(bid.params.tagId && haveSiteId && validFloor && validKeywords && (!isBanner ||
```
Comment (patmmccann) [inline modules/freestarBidAdapter.js:56]: are you really rejecting biid requests if the keywords are invalid? why not just drop the keywords

File: modules/freestarBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        ext: {}
+      };
+      rv.banner = {
+        format: _map(bid.sizes, ([width, height]) => ({w: width, h: height}))
+      };
+      const floor = getBidFloor(bid);
+      if (floor) {
+        rv.floor = floor;
+      }
+      if (bid.params.keywords) {
```
Comment (patmmccann) [inline modules/freestarBidAdapter.js:78]: please support reading keywords from ortb2 object

**Comment by patmmccann** [comment]
@nicgallardo looks good, just two requested changes with respect to keywords

File: modules/freestarBidAdapter.js (hunk 1 above)
Comment (nicgallardo) [reply modules/freestarBidAdapter.js:56]: Yes, we are validating just as we did in the SortableBidAdapter.

File: modules/freestarBidAdapter.js (hunk 2 above)
Comment (patmmccann) [reply modules/freestarBidAdapter.js:78]: you have to fix or delete support for keywords

File: modules/freestarBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/freestarBidAdapter.js:56]: this validation is quite aggressive, doesnt seem best for your company or your publisher partners

Comment (patmmccann) [reply modules/freestarBidAdapter.js:56]: also why not fallback to no floor when the floor is invalid?

**Pull request closed without merging**